internal/cmd/beta/network-interface/update/update.go

```diff
@@ -0,0 +1,242 @@
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to update the network interface %q?", model.NicId)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("create network area: %w", err)
+			}
+
+			return outputResult(p, model, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), networkIdFlag, "Network ID")
+	cmd.Flags().StringSlice(allowedAddressesFlag, nil, "List of allowed IPs")
+	cmd.Flags().StringToString(labelFlag, nil, "Labels are key-value string pairs which can be attached to a network-interface. A label can be provided with the format key=value and the flag can be used multiple times to provide a list of labels")
+	cmd.Flags().StringP(nameFlag, "n", "", "Network interface name")
+	cmd.Flags().BoolP(nicSecurityFlag, "b", defaultEnabled, "If this is set to false, then no security groups will apply to this network interface.")
+	cmd.Flags().StringSlice(securityGroupsFlag, nil, "List of security groups")
+
+	err := flags.MarkFlagsRequired(cmd, networkIdFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	nicId := inputArgs[0]
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	allowedAddresses := flags.FlagToStringSlicePointer(p, cmd, allowedAddressesFlag)
+	var allowedAddressesInner []iaas.AllowedAddressesInner
+	if allowedAddresses != nil && len(*allowedAddresses) > 0 {
+		allowedAddressesInner = make([]iaas.AllowedAddressesInner, len(*allowedAddresses))
+		for k, v := range *allowedAddresses {
+			// linter issue
+			value := v
+			allowedAddressesInner[k].String = &value
+		}
+	}
+
+	// check name length <= 63 and regex must apply
+	name := flags.FlagToStringPointer(p, cmd, nameFlag)
+	if name != nil {
+		if len(*name) > 63 {
+			return nil, &errors.FlagValidationError{
+				Flag:    nameFlag,
+				Details: fmt.Sprintf("name %s is too long (maximum length is 63 characters)", *name),
+			}
+		}
+		nameRegex := regexp.MustCompile(nameRegex)
+		if !nameRegex.MatchString(*name) {
+			return nil, &errors.FlagValidationError{
+				Flag:    nameFlag,
+				Details: fmt.Sprintf("name %s didn't match the required regex expression %s", *name, nameRegex),
+			}
+		}
+	}
+
+	// check security groups size and regex
+	securityGroups := flags.FlagToStringSlicePointer(p, cmd, securityGroupsFlag)
+	if securityGroups != nil && len(*securityGroups) > 0 {
+		securityGroupsRegex := regexp.MustCompile(securityGroupsRegex)
+		// iterate over them
+		for _, value := range *securityGroups {
+			if len(value) != 36 {
```

Comment:
It would be better if this `36` would be extracted to a `const` declared above, maybe even with a comment explaining that it is the current API limit